tests/foreman/ui/test_contenthost.py

```diff
@@ -422,6 +422,44 @@ def test_positive_remove_package_group(session, vm):
             assert not session.contenthost.search_package(vm.hostname, package)
 
 
+@tier3
+def test_actions_katello_host_package_update_timeout(session, vm):
+    """Check that Actions::Katello::Host::Package::Update task will time
+    out if goferd does not respond while attempting to update a package.
+
+    :id: 26f3ea2a-509a-4f3f-b5d7-d34b29ceb2cc
+
+    :BZ: 1651852
+
+    :expectedresults: Update task times out and error message is displayed.
```

Comment:
Just a curious question, where we are checking that the task is failing with time out? I see assumption here that we are assuming the if we see  `assert result['result'] == 'warning'` that means this is failed with time out error. 

@swadeley Do you think we need to add more assertion to ensure that is failed with timeout error only?

Reply:
ACK, will investigate. Thanks for looking.

Reply:
@swadeley any update?

Reply:
@JacobCallahan , I can get task duration from the table of tasks. I have been doing some testing with airgun, I found some widget needs updating (CSS code has changed). I might also be able to get task duration info using API, but I have been busy with other bugs. Will get back onto this in the morning.

Reply:
Hello, on the task details Error tab you can see: Exception
RuntimeError: Host did not respond within 20 seconds. The task has been cancelled. Is katello-agent installed and goferd running on the Host?

But I have no idea how to get that programatically (airgun or API).

Using airgun we can read:

In [3]: session.task.read('Update package for pzyeso3djzmen9an.lab.eng.rdu2.redhat.com walrus')                   
Out[3]:
{'menu': ['', 'Monitor', 'owyzdesp', 'Default Location'],
 'flash': [],
 'current_user': 'contenthost_sDesvfJ4ux',
 'account_menu': 'contenthost_sDesvfJ4ux',
 'logout': 'Log Out',
 'breadcrumb': 'Update package for pzyeso3djzmen9an.lab.eng.rdu2.redhat.com walrus',
 'task': {'name': 'N/A',
  'result': '',
  'triggered_by': 'Ended at:',
  'execution_type': 'Immediate',
  'start_at': '1 hour ago',
  'started_at': '1 hour ago',
  'ended_at': '1 hour ago',
  'state': 'State: stopped'}}

but that info does not prove that it failed with timeout error only.

Also can read:

In [7]: session.task.read_all('table')                                                                            
Out[7]:
{'table': [{'Select all rows': 'Select row',
   'Action': 'Update package for g5vni1949pbkhpui.lab.eng.rdu2.redhat.com walrus',
   'State': 'stopped',
   'Result': 'warning',
   'Started at': 'March 25, 2020, 11:41:27 AM',
   'Duration': '21 seconds',
   'Operation': 'Cancel'}]}

Reply:
I found this in the logs:

/var/log/foreman/production.log-20200326:2020-03-25T08:12:00 [E|bac|] Host did not respond within 20 seconds. The task has been cancelled. Is katello-agent installed and goferd running on the Host? (RuntimeError)

I regard that as proof. If we assert that test is present in the logs within 30 seconds of the "# Attempt to update fake package, check for warning" section, would @omkarkhatavkar approve this PR?

Reply:
@omkarkhatavkar is this sufficient?